doc/security.rst

```diff
@@ -0,0 +1,184 @@
+.. _recover from sudden power loss: https://www.youtube.com/watch?v=N8V0W0p3YBU&t=1115
+
+Whole-disk encryption protects against offline attacks as a side
+effect, assuming that the attacker cannot gain access to the secret
+key, because blocks cannot be decrypted/encrypted and thus files
+cannot be accessed and modified without the secret key.
+
+Whole-disk encryption is currently supported when installing to
+internal media using the ``image-installer`` command. See
+howtos/image-install.rst for information about this image installation
+method.
+
+The install step is necessary because in order to be more secure, a
+per-machine secret key should be used. Using the same key for multiple
+machines would increase the damage resulting from leaking that key.
+
+Currently, `LUKS`_ is used for the rootfs partition. Using LUKS has
+several advantages:
+
+* ``refkit-initramfs`` can detect whether a partition is encrypted
+
+* LUKS tools have sane default parameters for encryption
+
+* encryption parameters are stored in the partition
+  instead of having to be hard-coded in the installer and
+  initramfs, which also simplifies updates (initramfs will
+  always use the parameters that were used when creating
+  the partition)
+
+The downside is a higher risk for data loss, because the LUKS header
+blocks are required for mounting the partition. This could be avoided
+by setting up ``dm-crypt`` directly, which currently is not supported
+out-of-the-box.
+
+.. _LUKS: https://gitlab.com/cryptsetup/cryptsetup/blob/master/README.md
+
+To unlock the LUKS partition, a secret key is
+needed. ``image-installer`` and the corresponding
+:file:`meta-refkit/recipes-image/images/initramfs-framework-refkit-luks.bb`
+can use a fixed passphrase. To use that, invoke image-installer with::
+
+  FIXED_PASSWORD=yes image-installer
+
+Obviously this is not secure and only useful for testing. Something
+that may be secure enough for real products (depending on the threat
+model for the product) is storing the secret key in a TPM chip. At the
+moment, TPM 1.2 is supported. If ``image-installer`` finds a TPM
+device, it will take over ownership of the TPM and use it to store a
+randomly generated key in a fixed NVRAM slot. The assumption is that the
+new OS installation has full control over the TPM and does not need to
+share it with other OS installations.
+
+When booting, the initramfs will retrieve the key and then prevent
+further access to it, using the TPM ``READ_STCLEAR`` protection
+feature for this particular slot. The effect is that even if an
+attacker gains access to the TPM at runtime, reading the key again
+gets denied by the TPM chip itself. However, the secret key is of
+course still in use by the kernel and thus might be leaked from there
+and/or files modified at runtime by an attacker will be written to
```

Comment:
Is this better:

> The key remains in kernel memory and it is the kernel's responsibility to
> not leak it during a runtime attack. In addition, the kernel must prevent
> an attacker from modifying files at runtime, because such changes cannot
> be detected during the next boot.